normandy/base/api/authentication.py

```diff
@@ -81,11 +82,19 @@ def authenticate_header(self, request):
 
     @backoff.on_exception(backoff.constant, requests.exceptions.RequestException, max_tries=5)
     def fetch_oidc_user_profile(self, access_token):
+        cache_key = f"oidc-profile-{access_token}"
+        cached_response = cache.get(cache_key)
+
+        if cached_response:
+            return cached_response
+
         url = settings.OIDC_USER_ENDPOINT
         response = requests.get(url, headers={"Authorization": f"Bearer {access_token}"})
 
         if response.status_code == 200:
-            return response.json()
+            profile = response.json()
+            cache.set(cache_key, cache_key, 15)
```

Comment:
I just commented on this in the issue, but I think we shouldn't use a flat 15 second cache time. Instead I think we should read the rate limit information provided by the response (in the headers `X-RateLimitRemaining` and `X-RateLimit-Reset`), and adjust the cache based on that.